# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


# 1. Гіпотези і тип тесту
H₀: μ = 123.7

H₁: μ > 123.7

Тип тесту - правосторонній

# 2. Вибіркові статистики


In [1]:
import numpy as np

data = np.array([128, 135, 121, 142, 126, 151, 123])

n = len(data)
x_bar = np.mean(data)
s = np.std(data, ddof=1)

print("n =", n)
print("x̄ =", round(x_bar, 4))
print("s =", round(s, 4))

n = 7
x̄ = 132.2857
s = 10.9805


# 3. t-тест

In [2]:
from scipy import stats
import math

mu_0 = 123.7
alpha = 0.10

# t-статистика
t_stat = (x_bar - mu_0) / (s / math.sqrt(n))

# ступені свободи
df = n - 1

# критичне значення
t_crit = stats.t.ppf(1 - alpha, df)

# p-value (правосторонній тест)
p_value = 1 - stats.t.cdf(t_stat, df)

print("t =", round(t_stat, 4))
print("t_critical =", round(t_crit, 4))
print("p-value =", round(p_value, 4))

t = 2.0687
t_critical = 1.4398
p-value = 0.042


# 4. Чи відхиляємо гіпотезу  H₀  використовуючи p-value.

In [3]:
if p_value < alpha:
    print("Відхиляємо H0")
else:
    print("Не відхиляємо H0")

Відхиляємо H0


# 5. Висновок

На основі проведеного t-тесту отримано p-value, яке є меншим за рівень значущості α = 0.10. Тому нульову гіпотезу H₀ було відхилено.

Це означає, що є статистично значущі підстави вважати, що після редизайну сайту середній час до покупки користувачів збільшився.

# 6. 𝛼=0.05

Оскільки отримане p-value є меншим за 0.05, нульову гіпотезу H₀ також відхиляємо і на цьому рівні значущості.

Отже, зміна рівня значущості з α = 0.10 до α = 0.05 не впливає на прийняте рішення.

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

# 1. Гіпотези

Нульова гіпотеза:
H₀: p = 0.23

Альтернативна гіпотеза:
H₁: p > 0.23

Тип тесту - правосторонній

# 2. p̂

In [4]:
n = 1200
x = 311
p0 = 0.23
alpha = 0.03

p_hat = x / n

print("p̂ =", round(p_hat, 4))

p̂ = 0.2592


# 3. z-test

In [5]:
from statsmodels.stats.proportion import proportions_ztest

z_stat, p_value = proportions_ztest(x, n, value=p0, alternative='larger')

print("z =", round(z_stat, 4))
print("p-value =", round(p_value, 5))

z = 2.3058
p-value = 0.01056


# 4. Чи відхиляємо гіпотезу  H₀

In [6]:
if p_value < alpha:
    print("Відхиляємо H0")
else:
    print("Не відхиляємо H0")

Відхиляємо H0


# 5. Висновок

На основі проведеного z-тесту отримане p-value є меншим за рівень значущості α = 0.03, тому нульову гіпотезу H₀ було відхилено.

Це означає, що є статистично значущі підстави вважати, що після рекламної кампанії частка дорослих, які впізнають логотип, збільшилась і є більшою за 23%

# 6. Довірчий інтервал

In [7]:
from statsmodels.stats.proportion import proportion_confint

conf_int = proportion_confint(x, n, alpha=alpha, method='normal')

print("Інтервал:", round(conf_int[0], 4), "-", round(conf_int[1], 4))

Інтервал: 0.2317 - 0.2866


# 7. Інтерпретація довірчого інтервалу

З імовірністю 97% істинна частка дорослих, які впізнають логотип після рекламної кампанії, знаходиться в інтервалі від 0.2317 до 0.2866.

Оскільки значення 0.23 (історичне значення) знаходиться нижче нижньої межі довірчого інтервалу, це підтверджує висновок про те, що частка впізнаваності логотипу статистично значущо зросла після рекламної кампанії